feat(NKS): Add nodepool label & taints

gimmetm opened the pull request:
- resource_ncloud_nks_cluster
  - add hypervisor_code
  - lb_private_subnet_no, lb_public_subnet_no  change support
  - add ip_acl_default_action validtaion  : error on fin site
- datasource_ncloud_nks_cluster
  - add hypervisor_code
- resource_ncloud_nks_nodepool
  - subnet_no_list change support(only add)
  - add label
  - add taints 
- datasource_ncloud_nks_nodepool
  - add label
  - add taints 
- datasource_ncloud_nks_server_images
  - add hypervisor_code
- datasource_ncloud_nks_server_products
  - fix filter 
  - variable depth changed

mingyum-kim-n committed:
update: nks nodepool label & taints

youngmn commented on internal/service/nks/nks_cluster_data_source_test.go:
```diff
@@ -13,15 +14,18 @@ import (
+	}
 
-	resource.Test(t, resource.TestCase{
+	resource.ParallelTest(t, resource.TestCase{
```
AccTest 실행시 VPC 개수 제한으로 에러가 발생할 수 있어 병렬테스트를 단일 테스트로 변경하였습니다.
https://github.com/NaverCloudPlatform/terraform-provider-ncloud/pull/356

wonchulee replied:
This was changed for a mentee who couldn't release the number of VPC restrictions. Changing ParallelTest to plain test seems like a mistake that we've made before. I think It is okay to use ParallelTest here and we should provide another way to run it sequentially for some restricted environments someday.

youngmn replied:
테스트 코드도 병렬 테스트가 가능하도록 이미 반영 되어 있네요, close 하겠습니다.

youngmn commented on docs/data-sources/nks_node_pool.md:
```diff
@@ -25,8 +25,8 @@ data "ncloud_nks_node_pool" "node_pool"{
-* `node_pool_name` - (Required) The name of nodepool.
-* `cluster_uuid` - (Required) Cluster uuid.
+* `node_pool_name` - The name of nodepool.
+* `cluster_uuid` - Cluster uuid.
```
둘 다 required 로 유지되고 있는데, 문서에서 삭제 되었습니다.

gimmetm replied:
required  추가하였습니다.

youngmn commented on docs/resources/nks_cluster.md:
```diff
@@ -1,8 +1,3 @@
-subcategory: "Kubernetes Service"
----
-
-
```
각 문서마다 카테고리 설정이 추가되어 있습니다. 삭제되면 안되는 항목입니다.

gimmetm replied:
추가하였습니다.

youngmn commented on go.mod:
```diff
@@ -63,3 +63,5 @@ require (
 	google.golang.org/protobuf v1.31.0 // indirect
 )
+
+replace github.com/NaverCloudPlatform/ncloud-sdk-go-v2 v1.6.5 => ./../ncloud-sdk-go-v2-gimmetm
```
삭제 부탁 드립니다.

mingyum-kim-n committed:
fix: nks example typo

youngmn commented on internal/service/nks/nks_node_pools_data_source_test.go:
```diff
@@ -11,16 +12,18 @@ import (
-	region, clusterType, productType, k8sVersion := getRegionAndNKSType()
-
-	resource.Test(t, resource.TestCase{
+	resource.ParallelTest(t, resource.TestCase{
```
AccTest 실행시 VPC 개수 제한으로 에러가 발생할 수 있어 병렬테스트를 단일 테스트로 변경하였습니다.

youngmn commented on internal/service/nks/nks_node_pool_data_source_test.go:
```diff
@@ -12,16 +13,17 @@ import (
+	if err != nil {
+		t.Error(err)
+	}
+	resource.ParallelTest(t, resource.TestCase{
```
AccTest 실행시 VPC 개수 제한으로 에러가 발생할 수 있어 병렬테스트를 단일 테스트로 변경하였습니다.

youngmn commented on internal/service/nks/nks_kube_config_data_source_test.go:
```diff
@@ -12,15 +13,18 @@ import (
+	}
 
-	resource.Test(t, resource.TestCase{
+	resource.ParallelTest(t, resource.TestCase{
```
AccTest 실행시 VPC 개수 제한으로 에러가 발생할 수 있어 병렬테스트를 단일 테스트로 변경하였습니다.

youngmn reviewed (commented):
lint 에러가 발생하고 있는데, 확인 부탁드립니다.

wonchulee commented on internal/service/nks/nks_cluster_data_source_test.go:
```diff
@@ -46,62 +50,50 @@ func TestAccDataSourceNcloudNKSCluster(t *testing.T) {
+    audit                     = %[10]t
+  }
+  oidc {
+    issuer_url                = "https://keycloak.ncp.gimmetm.net/realms/nks"
```
It looks like an existing domain, Is it okay to use this URL for the common test?

gimmetm replied:
changed  to non-existing domain

wonchulee commented on internal/service/nks/nks_server_products_data_source.go:
This schema changes can cause error to the client who use detail attribute. So instead of deleting detail attribute, how about adding `Deprecated` in the schema definition?

mingyum-kim-n committed:
update:
- remove replace in go.mod
- add subcategory to nks_cluster docs
- rollback datasource.nks_node_pool Argument Reference
- datasource.ncloud_nks_server_products details deprecated

mingyum-kim-n committed:
fix: lint

wonchulee committed:
fix: Skip Acceptance tests when TF_ACC env is unset

wonchulee committed:
chore(nks): Update test url

mingyum-kim-n committed:
remove: NKS KVM Acc test cases

wonchulee committed:
chore(nks): Update example

wonchulee committed:
doc(nks): Update lb_public_subnet_no

mingyum-kim-n committed:
chore(nks): Update test url
